imnodes.cpp

```diff
@@ -2220,37 +2271,73 @@ void BeginNodeEditor()
 
     GImNodes->ImNodesUIState = ImNodesUIState_None;
 
-    GImNodes->MousePos = ImGui::GetIO().MousePos;
-    GImNodes->LeftMouseClicked = ImGui::IsMouseClicked(0);
-    GImNodes->LeftMouseReleased = ImGui::IsMouseReleased(0);
-    GImNodes->LeftMouseDragging = ImGui::IsMouseDragging(0, 0.0f);
-    GImNodes->AltMouseClicked =
-        (GImNodes->Io.EmulateThreeButtonMouse.Modifier != NULL &&
-         *GImNodes->Io.EmulateThreeButtonMouse.Modifier && GImNodes->LeftMouseClicked) ||
-        ImGui::IsMouseClicked(GImNodes->Io.AltMouseButton);
-    GImNodes->AltMouseDragging =
-        (GImNodes->Io.EmulateThreeButtonMouse.Modifier != NULL && GImNodes->LeftMouseDragging &&
-         (*GImNodes->Io.EmulateThreeButtonMouse.Modifier)) ||
-        ImGui::IsMouseDragging(GImNodes->Io.AltMouseButton, 0.0f);
-    GImNodes->AltMouseScrollDelta = ImGui::GetIO().MouseWheel;
-    GImNodes->MultipleSelectModifier =
-        (GImNodes->Io.MultipleSelectModifier.Modifier != NULL
-             ? *GImNodes->Io.MultipleSelectModifier.Modifier
-             : ImGui::GetIO().KeyCtrl);
-
     GImNodes->ActiveAttribute = false;
+    GImNodes->IsHovered = false;
 
     ImGui::BeginGroup();
     {
-        ImGui::PushStyleVar(ImGuiStyleVar_FramePadding, ImVec2(1.f, 1.f));
-        ImGui::PushStyleVar(ImGuiStyleVar_WindowPadding, ImVec2(0.f, 0.f));
-        ImGui::PushStyleColor(ImGuiCol_ChildBg, GImNodes->Style.Colors[ImNodesCol_GridBackground]);
-        ImGui::BeginChild(
-            "scrolling_region",
-            ImVec2(0.f, 0.f),
-            true,
-            ImGuiWindowFlags_NoScrollbar | ImGuiWindowFlags_NoMove |
-                ImGuiWindowFlags_NoScrollWithMouse);
+        // Setup zoom context
+        ImVec2 canvas_size = ImGui::GetContentRegionAvail();
+        GImNodes->CanvasOriginalOrigin = ImGui::GetCursorScreenPos();
+        GImNodes->OriginalImgCtx = ImGui::GetCurrentContext();
+
+        // Copy config settings in IO from main context, avoiding input fields
+        memcpy(
+            (void*)&GImNodes->NodeEditorImgCtx->IO,
+            (void*)&GImNodes->OriginalImgCtx->IO,
+            offsetof(ImGuiIO, SetPlatformImeDataFn) +
+                sizeof(GImNodes->OriginalImgCtx->IO.SetPlatformImeDataFn));
+
+        GImNodes->NodeEditorImgCtx->IO.BackendPlatformUserData = nullptr;
+        GImNodes->NodeEditorImgCtx->IO.BackendRendererUserData = nullptr;
+        GImNodes->NodeEditorImgCtx->IO.IniFilename = nullptr;
+        GImNodes->NodeEditorImgCtx->IO.ConfigInputTrickleEventQueue = false;
+        GImNodes->NodeEditorImgCtx->IO.DisplaySize = ImMax(canvas_size / editor.ZoomScale, ImVec2(0, 0));
+        GImNodes->NodeEditorImgCtx->Style = GImNodes->OriginalImgCtx->Style;
+
```

Comment:
Scaled(Zoom) context does not need viewport option, nor docking.
So, we have to disable (only if a user enabled these flags).
Otherwise assertion failur in `ImGui::ErrorCheckNewFrameSanityChecks()`

```
GImNodes->NodeEditorImgCtx->IO.ConfigFlags -= ImGuiConfigFlags_ViewportsEnable;
GImNodes->NodeEditorImgCtx->IO.ConfigFlags -= ImGuiConfigFlags_DockingEnable;

```